### PR TITLE
Refactor request manager and add tests;

### PR DESCRIPTION
**Pull request by jotshin**
JIRA ticket: https://conichi.atlassian.net/browse/CNI-11072

## Description
- Refactor CNIRequestManager
- Add tests for CNIRequestManager and CNIBookingManager
- Add snapshot tests for HTTP request. The snapshot tests are conducted by [this library](https://github.com/pointfreeco/swift-snapshot-testing), and it can not only be used for UI testing but also text based data, e.g. request, JSON, etc.

### CODE REVIEW

File: CNI-ItinerariesTests/CNIBookingManagerTests.swift [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                    subject.getBookingsWith(success: { _ in }, failure: { (error) in
+                        resultError = error
+                    })
+                    expect(resultError).toEventually(matchError(CNIHttpError.unknownError))
+                }
+            }
+        }
+        
+        describe(".postBookingWith") {
+            it("uses bookings endpoint and GET method") {
```
Comment (David-Henner) [inline CNI-ItinerariesTests/CNIBookingManagerTests.swift]: `POST method`

File: CNI-ItinerariesTests/CNIBookingManagerTests.swift [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                    subject.getBookingsWith(success: { _ in }, failure: { (error) in
+                        resultError = error
+                    })
+                    expect(resultError).toEventually(matchError(CNIHttpError.unknownError))
+                }
+            }
+        }
+        
+        describe(".deleteBookingWith") {
+            it("uses bookings endpoint and GET method") {
```
Comment (David-Henner) [inline CNI-ItinerariesTests/CNIBookingManagerTests.swift]: `DELETE method`

File: CNI-ItinerariesTests/CNIBookingManagerTests.swift (hunk 1 above)
Comment (jotshin) [reply CNI-ItinerariesTests/CNIBookingManagerTests.swift]: Nice catch. Updated in 4caf015.

File: CNI-ItinerariesTests/CNIBookingManagerTests.swift (hunk 2 above)
Comment (jotshin) [reply CNI-ItinerariesTests/CNIBookingManagerTests.swift]: Thanks and updated in 4caf015.

**Review by David-Henner — APPROVED** [review]

**Pull request merged**